pkg/docker/config/config.go

```diff
@@ -0,0 +1,271 @@
+)
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, registry string) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return errors.Wrapf(err, "error getting path to auth file")
+	}
+	encodedCreds := encodeUserAndPass(username, password)
+	return addAuthToJSON(registry, encodedCreds, path)
+}
+
+// GetAuthentication returns the value of the registry url stored in
+// the auth.json file
+func GetAuthentication(ctx *types.SystemContext, registry string) (string, string, error) {
+	if ctx != nil && ctx.DockerAuthConfig != nil {
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
+	}
+
+	// Check containers-storage auth.json file first
+	path, err := getPathToAuth()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "error getting path to auth file")
+	}
+	auth, err := findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	if auth != "" {
+		return decodeDockerAuth(auth)
+	}
+
+	// check .docker/config.json as secondary option
+	path = filepath.Join(homedir.Get(), dockerCfg, dockerCfgFileName)
+	auth, err = findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	if auth != "" {
+		decodeDockerAuth(auth)
+	}
+
+	return "", "", nil
+}
+
+// RemoveAuthentication deletes the credentials stored in auth.json
+func RemoveAuthentication(registry string) (string, error) {
+	path, err := getPathToAuth()
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	auths, err := readJSONFile(path)
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading JSON file %q", path)
+	}
+	if _, ok := auths.AuthConfigs[registry]; ok {
+		delete(auths.AuthConfigs, registry)
+	} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+		delete(auths.AuthConfigs, normalizeRegistry(registry))
+	} else {
+		return "", ErrNotLoggedIn
+	}
+
+	file, err := os.OpenFile(path, os.O_RDWR, 0755)
+	if err != nil {
+		return "", errors.Wrapf(err, "error opening file %q", path)
+	}
+	defer file.Close()
+
+	newData, err := json.MarshalIndent(auths, "", "\t")
+	if err != nil {
+		return "", errors.Wrapf(err, "error marshaling JSON %q", path)
+	}
+
+	if err = ioutil.WriteFile(path, newData, 0755); err != nil {
+		return "", errors.Wrapf(err, "error writing to file %q", path)
+	}
+	return registry, nil
+}
+
+// getPath gets the path of the auth.json file since the user can set their own path
+// by setting CRIO_AUTH_PATH
+// If CRIO_AUTH_PATH is not set, the default path is ~/.config/containers/auth.json
+func getPathToAuth() (string, error) {
+	if os.Getenv("CRIO_AUTH_PATH") != "" {
+		return filepath.Join(os.Getenv("CRIO_AUTH_PATH"), authCfgFileName), nil
+	}
+	dir := filepath.Join(homedir.Get(), authCfg)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err = os.Mkdir(dir, 0600); err != nil {
```

Comment:
Looking at the test failure, this might also need to create  ~/.config, probably with 0755 permissions .

Actually 0600 for a directory should make the directory unusable; that should be 0700.